ztp/siteconfig-generator/siteConfig/siteConfigBuilder.go

```diff
@@ -75,12 +76,16 @@ func (scbuilder *SiteConfigBuilder) validateSiteConfig(siteConfigTemp SiteConfig
 		if clusters[siteConfigTemp.Metadata.Name+"/"+cluster.ClusterName] {
 			return errors.New("Error: Repeated Cluster Name " + siteConfigTemp.Metadata.Name + "/" + cluster.ClusterName)
 		}
-		if cluster.NetworkType != "OpenShiftSDN" && cluster.NetworkType != "OVNKubernetes" {
-			return errors.New("Error: networkType must be either OpenShiftSDN or OVNKubernetes " + siteConfigTemp.Metadata.Name + "/" + cluster.ClusterName)
+		// will verify if installConfigOvverrides is a valid json if not empty
+		if cluster.InstallConfigOverrides != "" {
+			if !json.Valid([]byte(cluster.InstallConfigOverrides)) {
+				return errors.New("Error: Invalid json parameter set at installConfigOverride for " + siteConfigTemp.Metadata.Name + "/" + cluster.ClusterName)
+			}
+			siteConfigTemp.Spec.Clusters[id].NetworkType = cluster.InstallConfigOverrides
+		} else {
+			siteConfigTemp.Spec.Clusters[id].NetworkType = "{\"networking\":{\"networkType\":\"" + cluster.NetworkType + "\"}}"
```

Comment:
A user could include both networkType and installConfigOverrides in the same SiteConfig. I think in that case the expected behavior would be to merge the JSON content. 
If the user enters this in the siteconfig:
```
installConfigOverrides: |
  {
    "features" : {
      "first" : "abc",
      "second" : "def"
    }
  }
  networkType: OVNKubernetes
```

The resulting JSON would be:
```
{
  "features" : {
    "first" : "abc",
    "second" : "def"
  },
  "networking" : {
    "networkType": "OVNKubernetes"
  }
}
```

Reply:
@imiller0 fixed as suggested, added additional unit test.